policy-test/src/lib.rs

```diff
@@ -0,0 +1,101 @@
+#![deny(warnings, rust_2018_idioms)]
+#![forbid(unsafe_code)]
+
+pub mod admission;
+
+use linkerd_policy_controller_k8s_api::{self as api};
+use rand::Rng;
+use tracing::Instrument;
+
+/// Runs a test with a random namespace that is deleted on test completion
+pub async fn with_temp_ns<F, Fut>(test: F)
+where
+    F: FnOnce(kube::Client, String) -> Fut,
+    Fut: std::future::Future<Output = ()> + Send + 'static,
+{
+    let _tracing = init_tracing();
+
+    let namespace = {
+        let rng = &mut rand::thread_rng();
+        let sfx = (0..6)
+            .map(|_| rng.sample(LowercaseAlphanumeric) as char)
+            .collect::<String>();
+        format!("linkerd-policy-test-{}", sfx)
```

Comment:
nit/TIOLI: we could also include `thread::current().name()` in this string? if the tests are running in multi-threaded mode (which they do by default), each thread's name will be set to the name of the test, so this gives us a nice way to get the test's name. it could be useful to be able to see which test created which namespace.

the random characters are still probably worth keeping so that we don't hit weird errors if the tests are run multiple times against the same cluster...

Reply:
I like this idea but I'm going to leave it as a TODO for now. specifically, this will omit the actual test module name, so having multiple namespaces called _valid_ is still going to be confusing. Also, these namespaces are extremely short lived, and can be correlated via logs if really necessary.